src/js/components/Grommet/Grommet.js

```diff
@@ -19,26 +20,32 @@ class Grommet extends Component {
   static displayName = 'Grommet';
 
   static getDerivedStateFromProps(nextProps, prevState) {
-    const { dir, theme = {}, themeMode } = nextProps;
+    const {
+      background: backgroundProp,
+      dir,
+      theme = {},
+      themeMode,
+    } = nextProps;
     const { theme: stateTheme, themeProp, themeModeProp } = prevState;
 
     const nextTheme = deepMerge(baseTheme, theme);
     if (!stateTheme || theme !== themeProp || themeMode !== themeModeProp) {
       const {
-        colors: { background },
+        colors: { background: themeBackground },
       } = nextTheme.global;
-      // Determine whether to start in dark or light mode.
-      if (typeof background === 'object') {
-        // background is an object, use themeMode, theme default
-        // or first key
-        const color =
-          background[
-            themeMode || nextTheme.defaultMode || Object.keys(background)[0]
-          ];
-        nextTheme.dark = color ? colorIsDark(color) : false;
-      } else if (nextTheme.dark === undefined) {
-        nextTheme.dark = (background && colorIsDark(background)) || false;
-      }
+
+      // get initial value for dark so we can normalize background color
+      nextTheme.dark = (themeMode || theme.defaultMode) === 'dark';
+      const color = normalizeColor(
+        backgroundProp || themeBackground,
+        nextTheme,
+      );
+
+      // after normalizing, we want to check again if the background color
+      // is dark in case it differs from the previous assignment
```

Comment:
Is it necessary to have two assignments? can we normalize and then just assign once?

Reply:
It is necessary that we do it twice. We have to handle two cases:
1. caller passes in a color object or a color name that resolves an object. In this case, we want to set `dark` as line 38 shows. The second set, in line 46, is a no-op.
2. caller passes a specific color value or a color name that resolve to a specific color value. In this case, we want dark to be set based on that color, which line 46 will do.
The double set of `dark` allows us to handle both cases here without having to duplicate color object + name + dark mode detection code here that is already in `normalizeColor` and `backgroundIsDark`.
Maybe we should just put all of this in a comment, to be more verbose.

Reply:
Thanks for the detailed explanation. I will transfer this into a comment for clarity.